lib/promoted/ruby/client.rb

```diff
@@ -5,67 +5,103 @@
 module Promoted
   module Ruby
     module Client
-      class Error < StandardError; end
-      attr_accessor :options
-      BASE_URL          = "http://wh12.lvh.me:3000"
-      DELIVERY_ENDPOINT = "#{BASE_URL}/deliver"
-      LOGGING_ENDPOINT  =  "#{BASE_URL}/log_request"
-
-      def self.send_request payload, endpoint=nil
-        endpoint ||= BASE_URL
-        response = Faraday.post(endpoint) do |req|
-          req.headers['Content-Type'] = 'application/json'
-          req.body                    = payload.to_json
+
+      DEFAULT_DELIVERY_TIMEOUT_MILLIS = 3000
+      DEFAULT_METRICS_TIMEOUT_MILLIS = 250
+
+      class PromotedClient
+  
+        class Error < StandardError; end
+
+        attr_reader :perform_checks, :only_log, :uuid, :now_millis, :delivery_timeout_millis, :metrics_timeout_millis, :should_apply_treatment
```

Comment:
side-comment - this restructuring makes sense.

Reply:
Thanks :)